pnc-rest/src/main/java/org/jboss/pnc/rest/endpoint/ProjectEndpoint.java

```diff
@@ -59,8 +62,10 @@ public Response createNew(@NotNull @Valid ProjectRest projectRest, @Context UriI
     @ApiOperation(value = "Updates an existing Project")
     @PUT
     @Path("/{id}")
-    public Response update(@ApiParam(value = "Project id", required = true) @PathParam("id") Integer productId,
+    public Response update(@ApiParam(value = "Project id", required = true) @PathParam("id") Integer id,
             @NotNull @Valid ProjectRest projectRest, @Context UriInfo uriInfo) {
+        validateIdIsNull(projectRest);
```

Comment:
I thought about using bean validation for these checks, but I wasn't sure how well it would work because the validations are situational, i.e. in this case you want the ids to be null, but in other cases you might require the id to be filled in.  I wasn't sure how to implement a conditional validation like that using annotations.